crates/uv/tests/pip_compile.rs

```diff
@@ -7811,6 +7811,71 @@ fn no_version_for_direct_dependency() -> Result<()> {
     Ok(())
 }
 
+/// Compile against a dedicated platform, which may differ from the current platform.
```

Comment:
It looks like the test here doesn't change based on the platform? Or maybe my eyes missed it. Can we add a more "exciting" test where there is a change based on the platform?

Reply:
I'll add some more though. Do you have good examples from your work?

Reply:
Oh nice! I've mostly been working with `anyio`, although I think that's mostly dependent on `python_version`. Otherwise I made up my own packse scenarios.